### PR TITLE
Vb 3942 non association checks not filtering out affected visit sessions update

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?
Visits being checked for different prisons

## What is the intent behind these changes?
Live issue

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/prison/api/OtherPrisonerDetails.kt [hunk 1]
```diff
@@ -6,4 +6,7 @@ import io.swagger.v3.oas.annotations.media.Schema
 class OtherPrisonerDetails(
   @Schema(description = "The prisoner number", example = "G0135GA", required = true)
   val prisonerNumber: String,
+
+  @Schema(description = "The prison code", example = "MDI", required = false)
+  val prisonId: String? = null,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/prison/api/OtherPrisonerDetails.kt:11]: I would do non optional

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/helper/EntityTestBuilder.kt [hunk 2]
```diff
@@ -29,6 +29,57 @@ import java.time.LocalDate
...
+  policyNoticeDaysMax: Int = 28,
+  maxTotalVisitors: Int = 6,
+  maxAdultVisitors: Int = 3,
+  maxChildVisitors: Int = 3,
+  adultAgeYears: Int = 18,
+  isActive: Boolean = true,
+): Prison {
+  return Prison(code = prisonCode, active = isActive, policyNoticeDaysMin, policyNoticeDaysMax, maxTotalVisitors, maxAdultVisitors, maxChildVisitors, adultAgeYears)
+}
+fun sessionTemplate(
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/helper/EntityTestBuilder.kt:44]: space please

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/prison/api/OtherPrisonerDetails.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/prison/api/OtherPrisonerDetails.kt:11]: should I remove this check for now. will add it once confirmed

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/prison/api/OtherPrisonerDetails.kt]: removed the check for now. Will push this change through, do some more checks and then push this through.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/helper/EntityTestBuilder.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/helper/EntityTestBuilder.kt:45]: Updated

**Review by adaviesMOJ — APPROVED** [review]

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**